rpc/types_test.go

```diff
@@ -153,3 +153,33 @@ func TestBlockNumberOrHash_WithNumber_MarshalAndUnmarshal(t *testing.T) {
 		})
 	}
 }
+
+func TestBlockNumberOrHash_WithNumber_StringAndUnmarshal(t *testing.T) {
+	tests := []struct {
+		name  string
+		value BlockNumberOrHash
+	}{
+		{"max", BlockNumberOrHashWithNumber(math.MaxInt64)},
+		{"pending", BlockNumberOrHashWithNumber(PendingBlockNumber)},
+		{"latest", BlockNumberOrHashWithNumber(LatestBlockNumber)},
+		{"earliest", BlockNumberOrHashWithNumber(EarliestBlockNumber)},
+		{"0x20", BlockNumberOrHashWithNumber(32)},
+		{"hash", BlockNumberOrHashWithHash(common.Hash{0xaa}, false)},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
These tests are pretty blazing fast, I should think -- just marshalling some bytes -- do we really need the extra level of test abstraction? Typically it's nice if we want to paralellize, but in this case maybe just run them sequentially:

```golang
for i, test := range tests {
   bnh := test.value
   ...
   t.Fatalf("test %d: cannot unmars...", i, ...)
````